pkg/execution/svccatadmission/internaldns.go

```diff
@@ -0,0 +1,103 @@
+package svccatadmission
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/atlassian/voyager/pkg/microsserver"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/internaldns"
+	"github.com/atlassian/voyager/pkg/servicecentral"
+	sc_v1b1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/pkg/errors"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InternalDNSAdmitFunc checks existing DNS alias ownership via micros server API, for both InternalDNS Services and Ingress Resources
+func InternalDNSAdmitFunc(ctx context.Context, microsServerClient *microsserver.Client, scClient serviceCentralClient, admissionReview admissionv1beta1.AdmissionReview) (*admissionv1beta1.AdmissionResponse, error) {
+
+	admissionRequest := admissionReview.Request
+
+	// Validate supported resource type
+	if admissionRequest.Resource != serviceInstanceResource || admissionRequest.Resource != ingressResourceType {
+		return nil, errors.Errorf("unsupported resource, got %v", admissionRequest.Resource)
+	}
+
+	serviceName := getServiceNameFromNamespace(admissionRequest.Namespace)
```

Comment:
We need to stop using this function and use namespace informers and look at labels instead. But not in this PR of course.

Reply:
Yes, and there is even a ticket to do this.